src/main/java/org/demo/service/cxbox/anysource/lov/LovDao.java

```diff
@@ -38,7 +44,24 @@ public void delete(final BusinessComponent bc) {
 
 	@Override
 	public Page<DictDTO> getList(final BusinessComponent bc, final QueryParameters queryParameters) {
-		return lovClient.getAll(bc).getBody();
+		return new PageImpl<>(getAllListDictDTO(bc));
+	}
+
+	private List<DictDTO> getAllListDictDTO(final BusinessComponent bc) {
+		List<DictDTO> listDictDTO = Optional.ofNullable(lovClient.getAll(bc))
+				.stream()
+				.map(ResponseEntity::getBody)
+				.filter(Objects::nonNull)
+				.map(RestResponsePage::toList)
+				.findFirst().orElse(null);
+
+		if (listDictDTO == null) {
+			return new ArrayList<>();
+		}
+		return listDictDTO.stream()
+				.skip((long) bc.getParameters().getPageNumber() * bc.getParameters().getPageSize())
```

Comment:
pagination should be done on microservice side. there is no reason to do this in memory - pagination must protect from out of memory, but your variant does not do this